heatmap-api/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+use serde::{Deserialize, Serialize};
+
+pub trait ToPartialString {
+    fn _to_partial_string(&self) -> String;
+}
+
+#[derive(Deserialize, Serialize, Clone, Copy, Debug, PartialEq)]
+pub enum ProductTypes {
+    #[serde(rename = "GRD")]
+    GroundRangeDetected,
+    #[serde(rename = "SLC")]
+    SingleLookComplex,
+    #[serde(rename = "OCN")]
+    Ocean,
+}
+
+impl ProductTypes {
+    pub fn from_string(string: &str) -> Result<Self, std::fmt::Error> {
+        match string {
+            "GRD" => Ok(ProductTypes::GroundRangeDetected),
+            "SLC" => Ok(ProductTypes::SingleLookComplex),
+            "OCN" => Ok(ProductTypes::Ocean),
+            _ => Err(std::fmt::Error),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Clone, Copy, Debug, PartialEq)]
+pub enum PlatformType {
+    #[serde(rename = "SA")]
+    Sentinel1A,
+    #[serde(rename = "SB")]
+    Sentinel1B,
+}
+
+impl PlatformType {
+    pub fn from_string(string: &str) -> Result<Self, std::fmt::Error> {
+        match string {
+            "SA" => Ok(PlatformType::Sentinel1A),
+            "SB" => Ok(PlatformType::Sentinel1B),
+            _ => Err(std::fmt::Error),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Clone, Copy, Debug, PartialEq)]
+pub enum DataSensor {
+    #[serde(rename = "S")]
+    Sentinel,
+}
+
+impl DataSensor {
+    pub fn from_string(string: &str) -> Result<Self, std::fmt::Error> {
+        match string {
+            "S" => Ok(DataSensor::Sentinel),
+            _ => Err(std::fmt::Error),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct Filter {
+    pub product_type: Vec<ProductTypes>,
```

Comment:
I think we want to filter for product types for each platform type? So, maybe it makes more sense to create a dataset struct which holds a product type, an and platform type, (and a datasensor? and then the filter struct holds a vector of these, so that each product type is addressable to the platform type (IE you could have sentinel 1 slc data, but alos palsar grd data)